### DISCUSSION TITLE
JetBrains - IntelliJ, PhpStorm, WebStorm

### DISCUSSION THREAD

**Discussion by PwrSrg** [moalamri/vscode-inline-fold#66] [Ideas]
I **LOVE** this extension, but we use **JetBrains** products!

Can you **PLEASE** port this to the **JetBrains** platform?!? 🙏🏼

<a href="https://plugins.jetbrains.com/docs/intellij/developing-plugins.html">[Developing a Plugin | IntelliJ Platform Plugin SDK](https://plugins.jetbrains.com/docs/intellij/developing-plugins.html)</a>

**Comment by moalamri** [comment]
Hello @PwrSrg,
Thanks for the suggestion.
Back in 4th of Aug, JetBrains has issued a 1 All Products Pack license to help me port this extension to the JetBrains plugins marketplace.
After I looked into the editor's folding model, it seems that their IDEs lacks any APIs for horizontal positions, same goes with VS Code, it lacks the ability to mark folding horizontally.
The trick here is I used another API that falls below the editor's decoration it allows me to apply no-touch decorations to the text in the editor.
Which I couldn't find similar in the JetBrains SDK (unless I'm missing something)
So I finally gave up and was caught by work to this day, which I still find it difficult to port.

**Reply by PwrSrg** [reply]
Thanks for the detailed reply.  Too bad it won't work.  Hopefully someday! 😁

**Comment by denisborovikov** [comment]
Webstorm has a similar plugin https://plugins.jetbrains.com/plugin/19715-html-attribute-folder

**Reply by moalamri** [reply]
Nice 👍🏻 
I will mention the developer @Flyrell just to ask if it is natively supported or he also came up with a way to achieve that

**Reply by denisborovikov** [reply]
For those who are interested, I "made" a plugin using LivePlugin https://gist.github.com/denisborovikov/e5241c93f0dd9007b3fdb00d8c6e125b 

You can use it as a foundation and add your own rules. For now, it folds MUI `sx` prop.

**Reply by moalamri** [reply]
> For those who are interested, I "made" a plugin using LivePlugin https://gist.github.com/denisborovikov/e5241c93f0dd9007b3fdb00d8c6e125b
> 
> You can use it as a foundation and add your own rules. For now, it folds MUI `sx` prop.

Thanks 👍🏻 
```java
FoldRegion region = FoldingUtil.findFoldRegion(editor, start, end)
````
Are `start` and `end` located by column or line number?

**Reply by denisborovikov** [reply]
I have no idea. I don't know java or IntelliJ SDK, I just slightly modified one of the examples from the LivePlugin to do the job.
